gp_query.py

```diff
@@ -695,6 +697,145 @@ def _var_subst_chunk_result_ext(q_res, _sel_var_and_vars, _, **kwds):
 
 def _var_subst_res_update(res, update, **_):
     res += update
+    
+
+def useful_path_query(
```

Comment:
please rename to `deep_narrow_path_query`, helps to make the connection between query and mutation